common/xdc/nDCHistoryResender.go

```diff
@@ -131,7 +135,20 @@ func (n *NDCHistoryResenderImpl) SendSingleWorkflowHistory(
 
 	for historyIterator.HasNext() {
 		result, err := historyIterator.Next()
-		if err != nil {
+		switch err.(type) {
+		case nil:
+			// continue to process the events
+			break
+		case *shared.EntityNotExistsError:
+			// Case 1: the workflow pass the retention period
+			// Case 2: the workflow is corrupted
+			n.checkCurrentExecution(
```

Comment:
If the check here finds that the workflow is not corrupted, then does it mean the workflow has either passed retention period on the source cluster or the workflow/history is corrupted in the source cluster (for example a timer is fired on the corrupted workflow and generates some replication task)? and in this case do we still want to retry the replication task? Seems like we can just drop the task?

Reply:
Yes. we can drop the task